packages/statemanager/src/statefulBinaryTreeStateManager.ts

```diff
@@ -0,0 +1,802 @@
+    this._tree =
+      opts.tree ??
+      new BinaryTree({
+        hashFunction: this.hashFunction,
+        db: new MapDB<string, string | Uint8Array>(),
+        useRootPersistence: false,
+        cacheSize: 0,
+      })
+    this._debug = debugDefault('statemanager:binarytree:stateful')
+    this.originalStorageCache = new OriginalStorageCache(this.getStorage.bind(this))
+    this._caches = opts.caches
+    this.keccakFunction = keccak256
+    this.preStateRoot = new Uint8Array(32) // Initial state root is zeroes
+  }
+
+  /**
+   * Gets the account associated with `address` or `undefined` if account does not exist
+   * @param address - Address of the `account` to get
+   */
+  getAccount = async (address: Address): Promise<Account | undefined> => {
+    const elem = this._caches?.account?.get(address)
+    if (elem !== undefined) {
+      return elem.accountRLP !== undefined
+        ? createPartialAccountFromRLP(elem.accountRLP)
+        : undefined
+    }
+
+    const stem = getBinaryTreeStem(this.hashFunction, address, 0)
+
+    // First retrieve the account "header" values from the trie
+    const accountValues = await this._tree.get(stem, [
+      BinaryTreeLeafType.BasicData,
+      BinaryTreeLeafType.CodeHash,
+    ])
+
+    let account
+    if (accountValues[0] !== null && accountValues[0] !== undefined) {
+      const basicData = decodeBinaryTreeLeafBasicData(accountValues[0]!)
+      account = createPartialAccount({
+        version: basicData.version,
+        balance: basicData.balance,
+        nonce: basicData.nonce,
+        // Codehash is either untouched (i.e. null) or deleted (i.e. overwritten with zeros)
+        codeHash:
+          accountValues[1] === null ||
+          accountValues[1] === undefined ||
+          equalsBytes(accountValues[1], new Uint8Array(32))
+            ? KECCAK256_NULL
+            : accountValues[1],
+        codeSize: basicData.codeSize,
+        storageRoot: KECCAK256_NULL, // TODO: Add storage stuff
+      })
+    } else if (accountValues[1] === undefined || accountValues[1] === null) {
+      // account does not exist if both basic fields and codehash are undefined
+      if (this.DEBUG) {
+        this._debug(`getAccount address=${address.toString()} from DB (non-existent)`)
+      }
+      this._caches?.account?.put(address, account)
+    }
+
+    if (this.DEBUG) {
+      this._debug(`getAccount address=${address.toString()} stem=${short(stem)}`)
+    }
+    return account
+  }
+
+  public initBinaryTreeExecutionWitness(
+    _blockNum: bigint,
+    executionWitness?: BinaryTreeExecutionWitness | null,
+  ) {
+    if (executionWitness === null || executionWitness === undefined) {
+      const errorMsg = `Invalid executionWitness=${executionWitness} for initBinaryTreeExecutionWitness`
+      this._debug(errorMsg)
+      throw Error(errorMsg)
+    }
+
+    this.preStateRoot = hexToBytes(executionWitness.parentStateRoot) // set prestate root if given
```

Comment:
```suggestion
    this.preStateRoot = hexToBytes(executionWitness.parentStateRoot) // set prestate root
```
Prestate root isn't optional for a valid execution witness